src/charm.py

```diff
@@ -283,6 +283,40 @@ def _spring_boot_env(self) -> typing.Dict[str, str]:
             env[JAVA_TOOL_OPTIONS] = jvm_config
         return env
 
+    def _detect_mysql_capability(self) -> bool:
+        """Detect if the mysql lib is present in the spring-boot app.
+
+        Returns:
+            True if the mysql lib is present, False otherwise.
+
+        Raises:
+            TypeError: If the spring-boot app has a not recognizable type.
+        """
+        java_app = self._detect_java_application()
+        container = self._spring_boot_container()
+
+        if isinstance(java_app, BuildpackApplication):
+            try:
+                files = container.list_files("/workspace/BOOT-INF/lib")
+            except FileNotFoundError:
+                return False
+
+            return any(
+                file.name.startswith("mysql-connector-j-") and file.name.endswith(".jar")
+                for file in files
+            )
+
+        if isinstance(java_app, ExecutableJarApplication):
+            process = container.exec(["jar", "-t", "--file", java_app.executable_jar_path])
+            output, _ = process.wait_output()
+
+            return any(
+                filename.startswith("mysql-connector-j-") and filename.endswith(".jar")
+                for filename in (path.split("/")[-1] for path in output.splitlines())
+            )
+
+        raise TypeError("Unknown application type")
```

Comment:
Would it be better to define a custom error here rather than using an inbuilt error? That way our code knows whether it is handling our own errors or inbuilt Python errors?

Reply:
Code removed in the new version